TombEngine/Renderer/Renderer11DrawEffect.cpp

```diff
@@ -74,7 +76,95 @@ namespace TEN::Renderer
 		bool IsSoftParticle;
 	};
 
-	void Renderer11::DrawLightning(RenderView& view) 
+
+	void Renderer11::DrawTwogunLaser(RenderView& view)
+	{
+			byte r, g, b;
+
+			for (int i = 0; i < 2; i++)
+			{
+				TWOGUNINFO* tg = &twogun[i];
+
+				if (tg->life)
+				{
+					if (tg->fadein < 8)
+					{
+						r = (tg->r * tg->fadein) >> 3;
+						g = (tg->g * tg->fadein) >> 3;
+						b = (tg->b * tg->fadein) >> 3;
+					}
+					else if (tg->life < 16)
+					{
+						r = (tg->r * tg->life) >> 4;
+						g = (tg->g * tg->life) >> 4;
+						b = (tg->b * tg->life) >> 4;
+					}
+					else
+					{
+						r = tg->r;
+						g = tg->g;
+						b = tg->b;
+					}
+
+					LightningPos[0].x = tg->pos4.x;
+					LightningPos[0].y = tg->pos4.y;
+					LightningPos[0].z = tg->pos4.z;
+
+					memcpy(&LightningPos[1], tg, 48);
+
+					LightningPos[1].x = tg->pos1.x;
+					LightningPos[1].y = tg->pos1.y;
+					LightningPos[1].z = tg->pos1.z;
+			
+					for (int j = 0; j < 2; j++)
+					{
+						LightningPos[j].x -= tg->pos4.x;
+						LightningPos[j].y -= tg->pos4.y;
+						LightningPos[j].z -= tg->pos4.z;
+					}
+
+					CurlSpline(&LightningPos[0], LightningBuffer, tg);
```

Comment:
You are leaking one code part into another here. `CurlSpline` is defined in `lightning.cpp`, which has no relation to a renderer. If it's renderer-specific function, it must be contained in renderer class, not in game code.

Reply:
moved to renderer clas